test.py

```diff
@@ -936,11 +936,11 @@ def read_log(log_filename):
         return "===Error reading log {}===".format(e)
 
 
-def print_summary(failed_tests):
+def print_summary(failed_tests, options):
     if failed_tests:
         print("The following test(s) have failed: {}".format(
             palette.path(" ".join([t.name for t in failed_tests]))))
-        if not output_is_a_tty:
+        if options.verbose:
```

Comment:
I think this change is orthogonal to this series. @xemul did this output_is_a_tty thing because he explicitly didn't want to see long output on the terminal, and only see it when the output is redirected (as it is on Jenkins, or even if you just pipe it to "less"). I agree that it's weird that we have an explicit "--verbose" option and ignore it in this case, but I am not sure what were the original considerations. For example, it's easier to type "test.py | less" to see the full output instead of "test.py --verbose | less" which you'll need after your patch.
I would like @xemul to review this change.

Reply:
I understand the justification from @kostja , but I still don't see the use of the output of any test anywhere but in this test's log file. Even less I see the point in printing test log on the screen. The "test.py --verbose | less" seems very artificial, it will work even without this new --verbose option (let alone the fact that you have test log file anyway, why not lessing it?).

Reply:
By and large this change makes sense to me, I don't mind having it merged.